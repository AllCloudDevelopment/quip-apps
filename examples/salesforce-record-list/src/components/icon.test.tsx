// Copyright 2019 Quip

import React from "react";
import {mount} from "enzyme";

import Icon from "./icon";

describe("Icon component", () => {
    test("utility", () => {
        const allOn = mount(
            <Icon
                type="utility"
                object="edit"
                size="x-small"
                editable={true}
                sortable={true}
                rightInputIcon={true}/>);
        expect(allOn).toMatchSnapshot();
        const allOff = mount(
            <Icon
                type="utility"
                object="edit"
                size="x-small"
                editable={false}
                sortable={false}
                rightInputIcon={false}/>);
        expect(allOff).toMatchSnapshot();
    });
    test("standard", () => {
        const wrapper = mount(
            <Icon type="standard" object="agent_session" size="x-small"/>);
        expect(wrapper).toMatchSnapshot();
    });
    test("action", () => {
        const wrapper = mount(
            <Icon type="action" object="goal" size="x-small"/>);
        expect(wrapper).toMatchSnapshot();
    });
});
