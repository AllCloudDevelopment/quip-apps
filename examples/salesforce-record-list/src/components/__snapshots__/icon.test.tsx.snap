// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component action 1`] = `
<Icon
  editable={false}
  object="goal"
  rightInputIcon={false}
  size="x-small"
  sortable={false}
  type="action"
>
  <div
    className="icon slds-icon-action-goal"
  >
    <SLDSIcon
      assistiveText={Object {}}
      category="action"
      className="sldsIcon slds-icon slds-icon_x-small"
      colorVariant="default"
      name="goal"
      size="medium"
    >
      <span
        className="slds-icon_container slds-icon_container--circle slds-icon-action-goal"
      >
        <UtilityIcon
          aria-hidden="true"
          category="action"
          className="sldsIcon slds-icon slds-icon_x-small slds-icon"
          name="goal"
        >
          <svg
            aria-hidden="true"
            className="sldsIcon slds-icon slds-icon_x-small slds-icon"
            key="goal_action"
          >
            <use />
          </svg>
        </UtilityIcon>
      </span>
    </SLDSIcon>
  </div>
</Icon>
`;

exports[`Icon component standard 1`] = `
<Icon
  editable={false}
  object="agent_session"
  rightInputIcon={false}
  size="x-small"
  sortable={false}
  type="standard"
>
  <div
    className="icon slds-icon-standard-agent_session"
  >
    <SLDSIcon
      assistiveText={Object {}}
      category="standard"
      className="sldsIcon slds-icon slds-icon_x-small"
      colorVariant="default"
      name="agent_session"
      size="medium"
    >
      <span
        className="slds-icon_container slds-icon-standard-agent-session"
      >
        <UtilityIcon
          aria-hidden="true"
          category="standard"
          className="sldsIcon slds-icon slds-icon_x-small slds-icon"
          name="agent_session"
        >
          <svg
            aria-hidden="true"
            className="sldsIcon slds-icon slds-icon_x-small slds-icon"
            key="agent_session_standard"
          >
            <use />
          </svg>
        </UtilityIcon>
      </span>
    </SLDSIcon>
  </div>
</Icon>
`;

exports[`Icon component utility 1`] = `
<Icon
  editable={true}
  object="edit"
  rightInputIcon={true}
  size="x-small"
  sortable={true}
  type="utility"
>
  <div
    className="icon slds-icon-utility-edit"
  >
    <SLDSIcon
      assistiveText={Object {}}
      category="utility"
      className="sldsIcon slds-icon slds-icon_x-small slds-is-sortable__icon slds-button__icon_edit slds-input__icon slds-input__icon_right slds-icon-text-default"
      colorVariant="default"
      name="edit"
      size="medium"
    >
      <span>
        <UtilityIcon
          aria-hidden="true"
          category="utility"
          className="sldsIcon slds-icon slds-icon_x-small slds-is-sortable__icon slds-button__icon_edit slds-input__icon slds-input__icon_right slds-icon-text-default slds-icon slds-icon-text-default"
          name="edit"
        >
          <svg
            aria-hidden="true"
            className="sldsIcon slds-icon slds-icon_x-small slds-is-sortable__icon slds-button__icon_edit slds-input__icon slds-input__icon_right slds-icon-text-default slds-icon slds-icon-text-default"
            key="edit_utility"
          >
            <use />
          </svg>
        </UtilityIcon>
      </span>
    </SLDSIcon>
  </div>
</Icon>
`;

exports[`Icon component utility 2`] = `
<Icon
  editable={false}
  object="edit"
  rightInputIcon={false}
  size="x-small"
  sortable={false}
  type="utility"
>
  <div
    className="icon slds-icon-utility-edit"
  >
    <SLDSIcon
      assistiveText={Object {}}
      category="utility"
      className="sldsIcon slds-icon slds-icon_x-small slds-icon-text-default"
      colorVariant="default"
      name="edit"
      size="medium"
    >
      <span>
        <UtilityIcon
          aria-hidden="true"
          category="utility"
          className="sldsIcon slds-icon slds-icon_x-small slds-icon-text-default slds-icon slds-icon-text-default"
          name="edit"
        >
          <svg
            aria-hidden="true"
            className="sldsIcon slds-icon slds-icon_x-small slds-icon-text-default slds-icon slds-icon-text-default"
            key="edit_utility"
          >
            <use />
          </svg>
        </UtilityIcon>
      </span>
    </SLDSIcon>
  </div>
</Icon>
`;
